server/datastore/mysql/vulnerabilities.go

```diff
@@ -21,32 +22,75 @@ func (ds *Datastore) ListVulnerabilities(ctx context.Context, opt fleet.VulnList
 			cm.cisa_known_exploit,
 			cm.published,
 			COALESCE(cm.description, '') AS description,
-			vhc.host_count
+			vhc.host_count,
+			vhc.updated_at as host_count_updated_at
 		FROM
 			vulnerability_host_counts vhc
 		LEFT JOIN cve_meta cm ON cm.cve = vhc.cve
 		LEFT JOIN operating_system_vulnerabilities osv ON osv.cve = vhc.cve
 		LEFT JOIN software_cve sc ON sc.cve = vhc.cve
 		WHERE vhc.host_count > 0
 		`
-	groupByAppend := ` GROUP BY 
+	freeSelectStmt := `
+		SELECT
+			vhc.cve,
+			MIN(COALESCE(osv.created_at, sc.created_at, NOW())) AS created_at,
+			COALESCE(osv.source, sc.source, 0) AS source,
+			vhc.host_count,
+			vhc.updated_at as host_count_updated_at
+		FROM
+			vulnerability_host_counts vhc
+		LEFT JOIN operating_system_vulnerabilities osv ON osv.cve = vhc.cve
+		LEFT JOIN software_cve sc ON sc.cve = vhc.cve
+		WHERE vhc.host_count > 0
+		`
+
+	// Choose the appropriate select statement based on EE or Free
+	var selectStmt string
+	if opt.IsEE {
+		selectStmt = eeSelectStmt
+	} else {
+		selectStmt = freeSelectStmt
+	}
+
+	// Define group by statements for EE and Free
+	eeGroupBy := ` GROUP BY 
 			vhc.cve, 
+			source,
 			cm.cvss_score, 
 			cm.epss_probability, 
 			cm.cisa_known_exploit, 
 			cm.published, 
 			description, 
 			vhc.host_count
 	`
+	freeGroupBy := " GROUP BY vhc.cve, source, vhc.host_count"
+
+	// Choose the appropriate group by statement based on EE or Free
+	var groupBy string
+	if opt.IsEE {
+		groupBy = eeGroupBy
+	} else {
+		groupBy = freeGroupBy
+	}
 
+	// Prepare arguments for the query
 	var args []interface{}
 	if opt.TeamID == 0 {
-		selectStmt = selectStmt + " AND vhc.team_id = 0"
+		selectStmt += " AND vhc.team_id = 0"
 	} else {
-		selectStmt = selectStmt + " AND vhc.team_id = ?"
+		selectStmt += " AND vhc.team_id = ?"
 		args = append(args, opt.TeamID)
 	}
 
+	if opt.KnownExploit {
+		selectStmt += " AND cm.cisa_known_exploit = 1"
```

Comment:
Will this work for the free group-by ?

Reply:
KnownExploit is an EE only field, so an error is returned at the service level in the Free version